src/Protocol/ProtocolRecognizer.cpp

```diff
@@ -1225,22 +222,132 @@ void cProtocolRecognizer::HandlePacketStatusRequest(void)
 
 	AString Response = JsonUtils::WriteFastString(ResponseValue);
 
-	cPacketizer Pkt(*this, pktStatusResponse);
-	Pkt.WriteString(Response);
+	VERIFY(a_Out.WriteVarInt32(GetPacketID(cProtocol::ePacketType::pktStatusResponse)));
+	VERIFY(a_Out.WriteVarUTF8String(Response));
 }
 
 
 
 
 
-void cProtocolRecognizer::HandlePacketStatusPing()
+/* Ping handler for unrecognised versions. */
+static void HandlePacketStatusPing(cClientHandle & a_Client, cByteBuffer & a_Buffer, cByteBuffer & a_Out)
 {
 	Int64 Timestamp;
-	if (!m_Buffer.ReadBEInt64(Timestamp))
+	if (!a_Buffer.ReadBEInt64(Timestamp))
 	{
 		return;
 	}
 
-	cPacketizer Pkt(*this, pktPingResponse);
-	Pkt.WriteBEInt64(Timestamp);
+	VERIFY(a_Out.WriteVarInt32(GetPacketID(cProtocol::ePacketType::pktPingResponse)));
+	VERIFY(a_Out.WriteBEInt64(Timestamp));
```

Comment:
`cPacketizer` also called `SendPacket`. Is that not relevant any more?

Reply:
I call sendpacket after handling all the pings, inside TryRecogniseProtocol

Reply:
BUT NO! SendPacket sends a length + data. I can't go telling it to send two data at once, it's not going to produce a valid packet! soz, i derped